go-controller/pkg/ovn/gateway.go

```diff
@@ -251,30 +344,36 @@ func (oc *DefaultNetworkController) gatewayInit(nodeName string, clusterIPSubnet
 			IPPrefix: entry.String(),
 			Nexthop:  drLRPIfAddr.IP.String(),
 		}
+		if gw.netInfo.IsSecondary() {
+			lrsr.ExternalIDs = map[string]string{
+				types.NetworkExternalID:  gw.netInfo.GetNetworkName(),
+				types.TopologyExternalID: gw.netInfo.TopologyType(),
+			}
+		}
 		p := func(item *nbdb.LogicalRouterStaticRoute) bool {
 			return item.IPPrefix == lrsr.IPPrefix && libovsdbops.PolicyEqualPredicate(item.Policy, lrsr.Policy) &&
 				util.SliceHasStringItem(updatedLogicalRouter.StaticRoutes, item.UUID)
 		}
-		err = libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(oc.nbClient, gatewayRouter, &lrsr, p,
+		err = libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(gw.nbClient, gatewayRouter, &lrsr, p,
 			&lrsr.Nexthop)
 		if err != nil {
 			return fmt.Errorf("failed to add a static route %+v in GR %s with distributed router as the nexthop, err: %v", lrsr, gatewayRouter, err)
 		}
 	}
 
-	if err := oc.addExternalSwitch("",
+	if err := gw.addExternalSwitch("",
 		l3GatewayConfig.InterfaceID,
 		nodeName,
 		gatewayRouter,
 		l3GatewayConfig.MACAddress.String(),
```

Comment:
we need unique mac per network?

Reply:
@dceara / @tssurya ? could you chime in ?

Reply:
@martinkennelly  No, we share the node IP too between the host and the GR (for UDN too), we should share the MAC too.  I think. :)

Reply:
+1 I think we can share the MAC